<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Kushal K Dey, Rahul Mazumder" />


<title>Make gene bedfiles using CorShrink-sparse</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CorShrink2</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jdblischak/workflowr">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Make gene bedfiles using CorShrink-sparse</h1>
<h4 class="author"><em>Kushal K Dey, Rahul Mazumder</em></h4>
<h4 class="date"><em>1/19/2019</em></h4>

</div>


<p><strong>Last updated:</strong> 2019-01-19</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:red;">✖</strong> <strong>R Markdown file:</strong> uncommitted changes </summary> The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20190114)</code> </summary></p>
<p>The command <code>set.seed(20190114)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/kkdey/CorShrink2/tree/79c8c873f0bc9dda73888ba1724eebc91cec7950" target="_blank">79c8c87</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/BEDFILES/
    Ignored:    data/GSEA/

Untracked files:
    Untracked:  analysis/gene_lists_corshrink_sparse.Rmd
    Untracked:  code/corshrink2_tissue_tissue.R
    Untracked:  code/make_gene_lists.R
    Untracked:  docs/figure/gene_lists_corshrink_sparse.Rmd/

Unstaged changes:
    Modified:   code/gene_names_to_ensembl.R

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<hr />
<pre class="r"><code>library(corrplot)</code></pre>
<pre><code>corrplot 0.84 loaded</code></pre>
<pre class="r"><code>library(CorShrink)
setwd(&quot;~/Documents/CorShrink2/&quot;)</code></pre>
<div id="load-data" class="section level2">
<h2>Load Data</h2>
<pre class="r"><code>################   Fitted correlation matrix from CorShrink-sparse  #########################
corshrink_mat = get(load(&quot;data/CorShrink2Sparse_all_genes.rda&quot;))

###############  Log CPM normalized gene expression data   #############################
person_tissue_genes = get(load(&quot;data/person_tissue_genes_voom.rda&quot;))
tissue_names = names(person_tissue_genes[1,,1])

################   Gene names from the GTEx analysis  ############################
gene_names_gtex = as.character(read.table(&quot;data/gene_names_GTEX_V6.txt&quot;)[,1])
gene_names_gtex = as.character(sapply(gene_names_gtex, function(z) return(strsplit(z, &quot;[.]&quot;)[[1]][1])))

###############  Gene name with their locations on chromosomes  #############################

list_genes_qc = read.table(&quot;data/list_genes_qc.txt&quot;, header = TRUE)
ensg_names_list_genes = as.character(sapply(as.character(list_genes_qc$ENSG),
                                        function(z) return(strsplit(z, &quot;[.]&quot;)[[1]][1])))</code></pre>
<pre class="r"><code>num_records = 200</code></pre>
</div>
<div id="high-intra-brain-correlations-genes" class="section level2">
<h2>High intra brain correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[(8:20), (8:20), m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/brain_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   91    33    47    31    12    22    23    31    31     4    62    43 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   15    15    24    40    28    36    55    32    25    15 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/brain_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="high-all-correlations-genes" class="section level2">
<h2>High all correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[,, m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/all_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   54    22    38    21     8    18    21    30    43     3    39    20 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   14     6    11    28    13    30    37    22    14    18 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/all_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="low-all-correlations-genes" class="section level2">
<h2>Low all correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[,, m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/no_tissue_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   77    26    31    39    10    20    34    29    39    11    39    48 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   21     6    21    28    26    32    29    26    15    28 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[10]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/no_tissue_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="low-brain-correlations-genes" class="section level2">
<h2>Low brain correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[(8:20),(8:20), m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.7))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/non_brain_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   73    32    45    32     8    21    17    23    51     2    32    29 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   27     5    16    43    21    19    37    36    21    35 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/non_brain_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="just-non-brain-correlations-genes" class="section level2">
<h2>Just non brain correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[(8:20), (8:20), m]
  temp1 &lt;- corshrink_mat[-(8:20),-(8:20), m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.25)) - (quantile(temp1[row(temp1) &gt; col(temp1)], 0.75))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/just_non_brain_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   74    24    36    33     7    11    24    36    42     3    37    36 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   17    12    22    34    14    23    36    28    20    20 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[1]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/just_non_brain_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="just-brain-correlations-genes" class="section level2">
<h2>Just brain correlations (genes)</h2>
<pre class="r"><code>tab &lt;- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp &lt;- corshrink_mat[(8:20), (8:20), m]
  temp1 &lt;- corshrink_mat[-(8:20),-(8:20), m]
  tab[m] &lt;-  (quantile(temp[row(temp) &gt; col(temp)], 0.25)) - (quantile(temp1[row(temp1) &gt; col(temp1)], 0.75))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = &quot;data/GSEA/just_brain_corr_genes_list.txt&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)</code></pre>
<pre><code>
 chr1 chr10 chr11 chr12 chr13 chr14 chr15 chr16 chr17 chr18 chr19  chr2 
   99    37    53    37    13    21    29    37    42    10    94    47 
chr20 chr21 chr22  chr3  chr4  chr5  chr6  chr7  chr8  chr9 
   18    14    27    46    30    36    50    38    27    27 </code></pre>
<pre class="r"><code>df = newlist[, c(&quot;chr&quot;, &quot;start.ensembl&quot;, &quot;end.ensembl&quot;)]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 &lt;- c(&quot;blue&quot;, &quot;white&quot;, &quot;red&quot;)
corrplot(as.matrix(corshrink_mat[,,imp_ids[1]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = &quot;td&quot;, tl.cex = 0.4, tl.col = &quot;black&quot;,
         rect.col = &quot;white&quot;,na.label.col = &quot;white&quot;,
         method = &quot;color&quot;, type = &quot;upper&quot;)</code></pre>
<p><img src="figure/gene_lists_corshrink_sparse.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>write.table(df, file = &quot;data/BEDFILES/just_brain_corr_genes_ENSEMBL_top_1000.bed&quot;,
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] CorShrink_0.1-6 corrplot_0.84  

loaded via a namespace (and not attached):
 [1] gmp_0.5-13.2      Rcpp_1.0.0        plyr_1.8.4       
 [4] compiler_3.5.1    git2r_0.23.0      CVXR_0.99-2      
 [7] workflowr_1.1.1   R.methodsS3_1.7.1 R.utils_2.7.0    
[10] iterators_1.0.10  tools_3.5.1       digest_0.6.18    
[13] bit_1.1-14        gtable_0.2.0      evaluate_0.12    
[16] lattice_0.20-35   Matrix_1.2-14     foreach_1.4.4    
[19] yaml_2.2.0        parallel_3.5.1    gridExtra_2.3    
[22] Rmpfr_0.7-1       ECOSolveR_0.4     stringr_1.3.1    
[25] knitr_1.20        glmnet_2.0-16     rprojroot_1.3-2  
[28] bit64_0.9-7       grid_3.5.1        R6_2.3.0         
[31] rmarkdown_1.10    reshape2_1.4.3    corpcor_1.6.9    
[34] ashr_2.2-23       magrittr_1.5      whisker_0.3-2    
[37] backports_1.1.2   codetools_0.2-15  htmltools_0.3.6  
[40] scs_1.1-1         MASS_7.3-50       stringi_1.2.4    
[43] doParallel_1.0.14 pscl_1.5.2        truncnorm_1.0-8  
[46] SQUAREM_2017.10-1 R.oo_1.22.0      </code></pre>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
