---
title: "Make gene bedfiles using CorShrink-sparse"
author: "Kushal K Dey, Rahul Mazumder"
date: "1/19/2019"
output: html_document
---

```{r warning=FALSE}
library(corrplot)
library(CorShrink)
```

## Load Data

```{r}
################   Fitted correlation matrix from CorShrink-sparse  #########################
corshrink_mat = get(load("data/CorShrink2Sparse_all_genes.rda"))

###############  Log CPM normalized gene expression data   #############################
person_tissue_genes = get(load("data/person_tissue_genes_voom.rda"))
tissue_names = names(person_tissue_genes[1,,1])

################   Gene names from the GTEx analysis  ############################
gene_names_gtex = as.character(read.table("data/gene_names_GTEX_V6.txt")[,1])
gene_names_gtex = as.character(sapply(gene_names_gtex, function(z) return(strsplit(z, "[.]")[[1]][1])))

###############  Gene name with their locations on chromosomes  #############################

list_genes_qc = read.table("data/list_genes_qc.txt", header = TRUE)
ensg_names_list_genes = as.character(sapply(as.character(list_genes_qc$ENSG),
                                        function(z) return(strsplit(z, "[.]")[[1]][1])))
```

```{r}
num_records = 200
```

## High intra brain correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[(8:20), (8:20), m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/brain_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/brain_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```

## High all correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[,, m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/all_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/all_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```




## Low all correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[,, m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.25))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/no_tissue_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[10]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/no_tissue_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```


## Low brain correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[(8:20),(8:20), m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.7))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/non_brain_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[100]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/non_brain_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```

## Just non brain correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[(8:20), (8:20), m]
  temp1 <- corshrink_mat[-(8:20),-(8:20), m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.25)) - (quantile(temp1[row(temp1) > col(temp1)], 0.75))
}
imp_ids = order(tab, decreasing = FALSE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/just_non_brain_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[1]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/just_non_brain_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```

## Just brain correlations (genes)

```{r}
tab <- array(0, dim(corshrink_mat)[3])
for(m in 1:dim(corshrink_mat)[3]){
  temp <- corshrink_mat[(8:20), (8:20), m]
  temp1 <- corshrink_mat[-(8:20),-(8:20), m]
  tab[m] <-  (quantile(temp[row(temp) > col(temp)], 0.25)) - (quantile(temp1[row(temp1) > col(temp1)], 0.75))
}
imp_ids = order(tab, decreasing = TRUE)[1:1000]
imp_genes = gene_names_gtex[imp_ids]
write.table(imp_genes[1:num_records], file = "data/GSEA/just_brain_corr_genes_list.txt",
           quote=FALSE, row.names = FALSE, col.names = FALSE)
matched_ids = match(imp_genes, ensg_names_list_genes)
newlist = list_genes_qc[matched_ids[!is.na(matched_ids)],]
table(newlist$chr)
df = newlist[, c("chr", "start.ensembl", "end.ensembl")]
df[,2] = df[,2] - 10000
df[,3] = df[,3] + 10000

col2 <- c("blue", "white", "red")
corrplot(as.matrix(corshrink_mat[,,imp_ids[1]]), diag = FALSE,
         col = colorRampPalette(col2)(200),
         tl.pos = "td", tl.cex = 0.4, tl.col = "black",
         rect.col = "white",na.label.col = "white",
         method = "color", type = "upper")


write.table(df, file = "data/BEDFILES/just_brain_corr_genes_ENSEMBL_top_1000.bed",
           quote=FALSE, row.names = FALSE, col.names = FALSE, sep = "\t")

```
